### PR TITLE
Fix 21530 - dtoh: Sanitize identifiers wrt. reserved C++ keywords

### PR DESCRIPTION
**Pull request by MoonlightSentinel**
Extends the check for C++ keywords to append an underscore instead of
raising an error whenever the actual name doesn't matter for binary
compatibility.

CC @kinke @ibuclaw

### CODE REVIEW

File: src/dmd/frontend.h [hunk 1]
```diff
@@ -3562,7 +3562,7 @@ class ToCppBuffer final : public Visitor
     void visit(AliasDeclaration* ad);
     void visit(Nspace* ns);
     void visit(CPPNamespaceDeclaration* ns);
-    void handleNspace(Dsymbol* namespace, Array<Dsymbol* >* members);
+    void handleNspace(Dsymbol* namespace_, Array<Dsymbol* >* members);
```
Comment (ibuclaw) [inline src/dmd/frontend.h:3565]: Could just not emit parameter names?  They are just superfluous information at best.

Comment (kinke) [reply src/dmd/frontend.h:3565]: Sanitizing here as well makes sense IMO, especially as all comments are stripped, so documentation is scarce already.

Comment (RazvanN7) [reply src/dmd/frontend.h:3565]: Hmmm, maybe it would be useful if we could generate the documentation for the C++ headers also?

**Review by RazvanN7 — APPROVED** [review]

**Comment by MoonlightSentinel** [comment]
(Rebased to resolve the merge conflicts)

**Pull request merged**